editor/src/clj/editor/defold_project.clj

```diff
@@ -1102,26 +1091,11 @@
             node-type (resource-node-type resource)
 
             load-tx-data
-            (cond
+            (when (some? creation-tx-data)
               ;; If we just created the resource node, that means the referenced
               ;; resource does not exist, or it would have already been created
               ;; during resource-sync.
-              (some? creation-tx-data)
-              (do
-                (when (and *load-cache* (not (contains? @*load-cache* node-id)))
-                  (swap! *load-cache* conj node-id))
-                (mark-node-file-not-found node-id resource))
-
-              ;; If we're in the process of loading, we can verify that the
-              ;; referenced resource node has been loaded before the node that
-              ;; refers to it. If it hasn't been loaded yet, this means the
-              ;; :dependencies-fn of the consumer-node is not reporting it as a
-              ;; dependency. This is a programming error, so we throw an Error
-              ;; instead of an Exception.
-              (and *load-cache* (not (contains? @*load-cache* node-id)))
-              (throw (Error. (format "Node of type %s refers to '%s', but it hasn't been loaded yet. Check its :dependencies-fn."
-                                     (g/node-type-kw (:basis evaluation-context) consumer-node)
-                                     (resource/proj-path resource)))))]
```

Comment:
Had to remove this check now that we're lazy. I verified we do not do this more than once by forcing the eduction.